keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,358 @@
+# KEP-3698: Multi-Network requirements
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Personas](#personas)
+    - [Cluster operator](#cluster-operator)
+    - [Application developer](#application-developer)
+  - [Terminology](#terminology)
+  - [User Stories](#user-stories)
+    - [Story #1](#story-1)
+    - [Story #2](#story-2)
+    - [Story #3](#story-3)
+    - [Story #4](#story-4)
+    - [Story #5](#story-5)
+    - [Story #6](#story-6)
+    - [Story #7](#story-7)
+    - [Story #8](#story-8)
+    - [Story #9](#story-9)
+  - [Requirements](#requirements)
+    - [Phase I (base API and reference in Pod)](#phase-i-base-api-and-reference-in-pod)
+    - [Phase II (scheduler, kubelet and API probing for PodNetwork)](#phase-ii-scheduler-kubelet-and-api-probing-for-podnetwork)
+    - [Phase III (basic Kubernetes features integration)](#phase-iii-basic-kubernetes-features-integration)
+    - [Phase IV (extended functionality)](#phase-iv-extended-functionality)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Today Kubernetes Networking is very straightforward and easy to achieve. Main
+requirement is to enable connectivity between the Pods in the cluster. That
+simple approach satisfies most of Kubernetes customers, but it is not sufficient
+for cases where more complex networking is required:
+* some applications leverage different isolated networks, exposed through
+  different interfaces
+* some leverage performance oriented interfaces (e.g. AF_XDP, memif, SR-IOV),
+  besides the regular management interface
+* others require support for specific protocols not yet supported in Kubernetes
+
+For such requirements we need a solution that allows a user to express this in
+the pod specification. This can be done by an external solution leveraging
+annotations, but having it incorporated in Kubernetes would be a much cleaner
+and safer approach that would allow better compatibility and consistency for
+workloads with these needs.
+
+This KEP is an entry level document for the whole Multi-Network endeavor. Here
+we will define a set of requirements to follow. Additionally, we will introduce
+a phased approach where each phase will have its own KEP with detailed design.
```

Comment:
I had some thoughts about shortening the summary up a bit:

```suggestion
One of the main requirements for Kubernetes networking today is to enable
connectivity between `Pods` in a cluster. This facility satisfies a large number of
Kubernetes users, but it isn't sufficient for some important cases:

* applications leveraging different isolated networks exposed through different
  interfaces
* applications leveraging performance-oriented interfaces (e.g. `AF_XDP`,
  `memif`, `SR-IOV`)
* applications requiring support for protocols not yet supported by Kubernetes

Here we propose a solution that enables multiple network interfaces to be
defined in the `Pod` specification so as to provide a standard and clear 
mechanism to implement these more complex configurations.
```

Only if you like them, if you don't no worries please feel free to resolve this comment at your discretion.